src/history.rs

```diff
@@ -8,6 +8,8 @@ pub struct History {
 
 impl History {
     pub fn new(path_buf: PathBuf) -> Self {
+        #[cfg(not(test))]
+        let path_buf = path_buf.canonicalize().unwrap();
```

Comment:
I'd prefer to not conditionally change internal logic like this to get the tests working. Personally I would

1) Setup a tempdir with the expected files for `crate::history::tests::sanity`, since that test requires both `History::new` and History::make_next` to canonicalize correctly
2) Maybe provide some `#[cfg(test)]` `History::new_abs()` constructor or something like that which asserts that it has been passed an absolute path and then the tests can be adapted to use that instead